sycl/include/CL/__spirv/spirv_types.hpp

```diff
@@ -129,10 +136,12 @@ enum class MatrixLayout : uint32_t {
 // information to SPIRV translator.
 // The long term solution would be to introduce a matrix type in Clang and use
 // it instead of this member.
-template <typename T, std::size_t R, std::size_t C, MatrixLayout U,
+template <typename T, std::size_t R, std::size_t C, MatrixUse U, MatrixLayout L,
```

Comment:
This matches the template arguments used for `joint_matrix` in matrix-tensorcore.hpp (the current CUDA impl), except that the ordering is a little different (MatrixUse is the second template param in matrix-tensorcore.hpp).  Trivial point, but we should align an ordering before merging the impls into a single `SYCL_EXT_ONEAPI_MATRIX` version.

Reply:
This is a good point. Currently, this is at this position because it is an optional argument for now (to keep backward compatibility with the previous API (no use argument)).
Once this API with "use" argument is stable enough that we can remove the non-use-API, we can revise the order. 
My personal preference is that it is probably a good idea to keep the arguments that might be become "optional"  in the future as last. Today, "use" is required but layout is definitely optional.

Reply:
I suggest not to modify the order of template parameters of internal SPIR-V joint matrix type representation. The reason is that after this patch: https://github.com/intel/llvm/pull/6535/ the clang started to generate opaque matrix type like this: `spirv.JointMatrixINTEL._half_10_2_0_0` and hence we can and will remove the array W/A. So it becomes crucial to keep the template parameter's order internally (it still can be changed in user-visible API).

Reply:
So, we will keep the Workaround in SPIRVRegularizeLLVMBase::adaptStructTypes for now. Since the W/A will be removed in the future, we'd better move the Use at the end of template param list, right?

Reply:
Yes